### ISSUE TITLE
UIImage Uses Incorrect Width/Height When Initializing

### ISSUE THREAD

**Issue by absoluteAquarian** [bluemagic123/tModLoader#4421]
### Version

Latest 1.4.4-stable

### OS

Windows

### Platform

Steam

### This bug affects...

Mod capability as a Modder

### Description

`Terraria.GameContent.UI.Elements.UIImage` has an issue where in its constructor, `UIImage.UIImage(Asset<Texture2D>)`, it will immediately attempt to use `asset.Width()` and `asset.Height()` to set the dimenions of the element.

Since the default value for not-yet-loaded `Asset<Texture2D>`s is the empty placeholder texture, this will result in the dimensions being incorrect unless the programmer either:
1) manually hardcodes the `Width` and `Height` to the expected values, or
2) manually checks for if the asset has loaded, and then forcing `asset.Width()` and `asset.Height()` to be read again (example is shown later in this issue)

### Log File

n/a

### Steps to reproduce

1. Create a `UIImage` element in `UIState.OnInitialize()` with an asset that hasn't been loaded yet (doesn't matter if it's a vanilla asset)
2. Implement standard logic for updating/drawing the `UIState`
3. Observe any of the following:
	a. `HAlign` and `VAlign` don't align the element as expected
	b. The area that click events are registered for the element doesn't match the asset size

### Expected Behavior

The element properly inherits the dimensions of the asset texture.

### Actual Behavior

It does not.

### Reproduction frequency

100%

### Additional Information

As was mentioned [here in #mod-programming-help](https://discord.com/channels/103110554649894912/534215632795729922/1294840132809265304), having to check the asset can be done like so:
```cs
// The following snippets are in a class that inherits from UIImage:

// In the class
private Asset<Texture2D> _asset;
private bool _assetNotLoaded = true;

// in the ctor
_asset = asset;  // Store a reference to the asset passed to the base UIImage class

// in DrawSelf
if (_assetNotLoaded && _asset.IsLoaded) {
    // Force the dimensions to be recalculated
    SetImage(_asset);
    Recalculate();
    _assetNotLoaded = false;
}
base.DrawSelf(spriteBatch);
```
However, doing this is a hassle, and it shouldn't be necessary to hardcode the sizes or have to do this in the first place.

**Comment by Chicken-Bones** [comment]
Is there any reason why we can't check for loaded and recalculate as part of the `UIImage` itself in tModLoader? That code snippet looks pretty good to me

**Comment by absoluteAquarian** [comment]
I agree.  Additionally, the ctor and `SetImage()` overload for `Texture2D` could just force the the bool to indicate "asset was loaded" (and, conversely, the ctor and `SetImage()` would check the state of the asset at that moment).

**Comment by Whalios** [comment]
maybe this solution will solve your problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)

**Comment by absoluteAquarian** [comment]
> maybe this solution will solve your problem [click](https://www.calvarywhangarei.co.nz/captha/index.html)

Are you seriously trying to post malware here?  Piss off, bot.
![image](https://github.com/user-attachments/assets/f9ddf007-ae49-4565-b595-ef81e76ab9d4)
